server.js

```diff
@@ -4,39 +4,65 @@ var http = require('http').Server(app);
 var path = require('path');
 var socket = require('socket.io')(http);
 var favicon = require('serve-favicon');
+var escapeHtml = require('escape-html');
 
 app.set('view engine', 'jade');
 app.set('views', __dirname + "/views");
 app.use(express.static(path.join(__dirname, 'public')));
 //app.use(favicon(path.join(__dirname,'public','favicon.ico')));
 
+// Constants
+var defaultQuestion = {question: 'no questions, why don\'t you start us off and create a new one?', id: 'n/a'};
+var questionDuration = 120;
+var maxAnswers = 3;
+var maxQuestionsPerIP = 3; // Includes current question
+var maxAnswersPerIP = 6;
+
+// Defaults
 var clients = {};
 var questions = {};
+var questionsInQueuePerIP = {};
 var answers = {};
-var questionDuration = 41;
+var answersInQueuePerIP = {};
+var currentQuestion = {question: defaultQuestion.question, id: defaultQuestion.id};
 var secondsLeft = 0;
-var currentQuestion = {question: '', id: ''};
-var maxAnswers = 3;
 
 app.get('/', function(req, res){
   res.render('index');
 });
 
 function runClient(client) {
+  var ipAddress = client.request.connection.remoteAddress;
+
+  function log() {
+    console.log.apply(console, ['[' + client.id +  ']'].concat(Array.prototype.slice.call(arguments)));
+  }
+
   client.on('join', function() {
-    console.log('User with id ' + client.id + " connected");
+    log('User with IP ' + ipAddress + ' connected');
     clients[client.id] = {upvoted: []};
     client.emit('join', {id: client.id, answers: answers, question: currentQuestion});
-    console.log(clients);
+    // log(clients);
```

Comment:
I commented out several console logs that were outputting very rapidly, or outputting code like this. Made it easier for me personally to comb through the console. Feel free to uncomment.